implementations/rust/ockam/ockam_api/src/nodes/registry.rs

```diff
@@ -1,31 +1,59 @@
 use crate::nodes::service::Alias;
 use ockam_core::compat::collections::BTreeMap;
-use ockam_core::{Address, Result, Route};
-use ockam_identity::{Identity, IdentityIdentifier};
-use ockam_vault::Vault;
-use std::sync::Arc;
+use ockam_core::{Address, Route};
+use ockam_identity::IdentityIdentifier;
+
+#[derive(Default)]
+pub(crate) struct SecureChannelRegistry {
+    channels: Vec<SecureChannelInfo>,
+}
+
+impl SecureChannelRegistry {
+    pub fn get_by_route(&self, route: &Route) -> Option<&SecureChannelInfo> {
+        self.channels.iter().find(|&x| x.route() == route)
+    }
+
+    pub fn get_by_addr(&self, addr: &Address) -> Option<&SecureChannelInfo> {
+        self.channels.iter().find(|&x| x.addr() == addr)
+    }
+
+    pub fn insert(
+        &mut self,
+        addr: Address,
+        route: Route,
+        authorized_identifiers: Option<Vec<IdentityIdentifier>>,
+    ) {
+        self.channels
+            .push(SecureChannelInfo::new(route, addr, authorized_identifiers))
```

Comment:
I would not allow duplicate address or route registrations.

Reply:
Collision check wouldn't hurt, however those addresses are randomly generated, so I'm not sure what should happen so that we end up with duplicated